test/NuGet.Clients.Tests/NuGet.VisualStudio.Common.Test/OutputConsoleLoggerTests.cs

```diff
@@ -49,5 +67,12 @@ void IDisposable.Dispose()
         {
             _outputConsoleLogger.Dispose();
         }
+
+        public Task DisposeAsync() => Task.CompletedTask;
+
+        public async Task InitializeAsync()
```

Comment:
After the derived test classes constructor, this `InitializeAsync` ensures that the semaphore is reset before beginning test execution.
Some tests, however, trigger the semaphore within the test themselves. That's why you'll see adhoc calls to `WaitForInitialization()` in addition to this one.